subsys/power/device_pm.c

```diff
@@ -192,9 +191,9 @@ void device_pm_enable(const struct device *dev)
 		dev->pm->dev = dev;
 		atomic_set(&dev->pm->fsm_state,
 			   DEVICE_PM_STATE_SUSPENDED);
-		k_work_init(&dev->pm->work, pm_work_handler);
+		k_work_init_delayable(&dev->pm->work, pm_work_handler);
 	} else {
-		k_work_submit(&dev->pm->work);
+		k_work_schedule(&dev->pm->work, K_USEC(0));
```

Comment:
`K_NO_WAIT`

Reply:
will change, that is the part that I'm not particularly happy, I have to use `k_work_delayable`  otherwise the thread is scheduled out and the notification is missed causing a dead lock. For now this seems to work, but I want to remove the need of the work queue.